frontend/src/devicelist/DevicesDataTable.jsx

```diff
@@ -0,0 +1,142 @@
+import React from "react";
+import { Box, TableContainer, Typography } from "@material-ui/core";
+
+import {
+  FilterList,
+  Search,
+  ExpandLess,
+  ExpandMore,
+  ArrowDownward,
+  Clear,
+  SaveAlt
+} from "@material-ui/icons";
+
+import MaterialTable, { MTableToolbar } from "material-table";
+
+import ChannelDetailsTable from "./ChannelDetailsTable";
+import StatusIndicator from "../general/StatusIndicator";
+import ActionMenu from "./ActionMenu";
+
+export default class DevicesDataTable extends React.Component {
+  getComponents() {
+    return {
+      Toolbar: (props) => (
+        <div className="lightestGrey">
+          <MTableToolbar {...props} />
+        </div>
+      )
+    };
+  }
+
+  getColumnInfo() {
+    return [
+      {
+        title: "Name",
+        field: "name"
+      },
+      {
+        title: "Serial Number",
+        field: "serialNumber"
+      },
+      {
+        title: "Status",
+        field: "status",
+        render: (rowData) => <StatusIndicator status={rowData.status} />,
+        lookup: {
+          Online: "Online",
+          Pending: "Pending",
+          Error: "Error",
+          Offline: "Offline"
+        }
+      },
+      {
+        title: "IP Address",
+        field: "ip"
+      },
+      {
+        title: "Actions",
+        field: "action",
+        filtering: false,
+        sorting: false,
+        render: (rowData) => <ActionMenu />,
+        align: "center"
+      }
+    ];
+  }
+
+  getDetailPanel() {
+    return [
+      {
+        icon: ExpandMore,
+        openIcon: ExpandLess,
+        tooltip: "Show Device Details",
+        render: (rowData) => {
+          return (
+            <div className="lightestGrey">
+              <Box margin={2}>
+                <Typography variant="h6">Channels</Typography>
+                {rowData.channels.map((channel) => {
+                  return (
+                    <ChannelDetailsTable
+                      channel={channel}
+                      key={`ch_${channel.id}_p${channel.port}`}
+                    />
+                  );
+                })}
+              </Box>
+            </div>
+          );
+        }
+      }
+    ];
+  }
+
+  getOptions() {
+    return {
+      toolbar: true,
+      search: true,
+      exportButton: true,
+      headerStyle: {
+        backgroundColor: "#f1f1f1",
+        fontWeight: "bold"
+        // position: 'sticky', top: 0
+      },
+      // maxBodyHeight: '650px',
+      // above lines commented out b/c scrollbar bug, see: https://github.com/mbrn/material-table/issues/780
```

Comment:
From what I can tell in the linked issue, these lines are CSS that allow us to have sticky headers. Why not uncomment them?

Reply:
There'll be a second scrollbar that makes it so that when that scrollbar is activated it no longer stays sticky. It's hard to explain but I could provide a GIF.

Reply:
>     * I noticed very few changes to the tests were made. Is this because the old tests still validate the new table properly? I'm worried that even though the coverage is high, we aren't actually validating that stuff is working properly.

Nevermind this point, I somehow missed the devicelist test file when reviewing >.<